crates/release_plz_core/src/command/update/updater.rs

```diff
@@ -809,14 +809,20 @@ fn get_package_files(
         .filter(|l| l != "Cargo.toml.orig" && l != ".cargo_vcs_info.json")
         .map(|l| {
             let is_crate_path_same_as_git_repo = crate_relative_path == "";
+            // Normalize path to handle symbolic links correctly.
+            let normalized = fs_utils::canonicalize_utf8(&crate_relative_path.join(l))?;
             if is_crate_path_same_as_git_repo {
-                l
+                let prefix = repository.directory();
+                let stripped = normalized
+                    .strip_prefix(prefix)
+                    .with_context(|| format!("failed to strip {prefix} from {normalized}"))?;
+                Ok(stripped.to_path_buf())
             } else {
-                crate_relative_path.join(l)
+                Ok(normalized)
```

Comment:
the comment of the function says "The paths are relative to the git repo root". If you see how this function is used you get why: [here](https://github.com/release-plz/release-plz/blob/69805e71822d126803f95245346e2a8da5b859a7/crates/release_plz_core/src/command/update/updater.rs#L765) we compare this set of files with the git output.

Here you are returning the normalized output, so this won't work, right? Or am I missing something?

btw thanks for sending this PR!

Reply:
Ah yes, good catch. I think we'll have to strip the prefix on both branches here instead of just the first one. That should be doable.

Reply:
Pushed a fix for this. Since we're recreating the absolute (canonicalized) path for each item in the loop, we have to strip the prefix each time and the check for `crate_relative_path == ""` isn't useful anymore.